ckanext/scheming/tests/test_dataset_display.py

```diff
@@ -1,10 +1,73 @@
 import pytest
 import six
-from ckantoolkit.tests.factories import Sysadmin, Dataset
+from ckan.lib.helpers import url_for
+from ckantoolkit.tests.factories import Sysadmin, User, Dataset
+from bs4 import BeautifulSoup
 
 
 @pytest.mark.usefixtures("clean_db")
 class TestDatasetDisplay(object):
+
+    core_resource = {
+        'name': 'Core Resource',
+        'url': 'http://example.com/camel.txt',
+        'resource_type': 'awesome-resource'
+    }
+    extra_resource = {
+        'name': 'Extra Resource',
+        'url': 'http://example.com/camel.txt',
+    }
+
+    def test_core_resources(self, app):
+        user = User()
+        dataset = Dataset(
+            user=user,
+            type='test-schema',
+            resources=[self.core_resource],
+        )
+        response = app.get(
+            url_for('dataset.read', id=dataset['name']),
+            extra_environ={'REMOTE_USER': six.ensure_str(user['name'])},
+        )
+        core_resources_container = BeautifulSoup(response.body)\
+            .select('.core-resources')[0]
+        assert self.core_resource['name'] in str(core_resources_container)
+
+    def test_extra_resources(self, app):
+        user = User()
+        dataset = Dataset(
+            user=user,
+            type='test-schema',
+            resources=[self.extra_resource],
+        )
+        response = app.get(
+            url_for('dataset.read', id=dataset['name']),
+            extra_environ={'REMOTE_USER': six.ensure_str(user['name'])},
+        )
+        extra_resources_container = BeautifulSoup(response.body)\
```

Comment:
Instead of breaking the line with `\` (which is not very popular and quite error prone) create a tmp var:
```
body = BeautifulSoup(response.body)
core_resources_div = body.select('.core-resources')[0]
```